### ISSUE TITLE
DeprecationWarning: 'get_engine' is deprecated 

### ISSUE THREAD

**Issue by mokadevcloud** [miguelgrinberg/Flask-Migrate#510]
**Describe the bug**
Hi, we have the DeprecationWarning below upon running `flask db upgrade`, in spite of using the latest versions of flask, flask-migrate, flask-sqlachemy etc:
```
Flask==2.2.3
Flask-APScheduler==1.12.4
flask-babel==3.0.1
Flask-Cors==3.0.10
Flask-Migrate==4.0.4
flask-redis==0.4.0
Flask-SocketIO==5.3.2
Flask-SQLAlchemy==3.0.3
```
Can you please help?

**Logs**
```
/home/abcd-xyz/migrations/env.py:25: DeprecationWarning: 'get_engine' is deprecated and will be removed in Flask-SQLAlchemy 3.1. Use 'engine' or 'engines[key]' instead.
  str(current_app.extensions['migrate'].db.get_engine().url).replace(
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
```

**Comment by miguelgrinberg** [comment]
This has already been addressed in the latest Flask-Migrate releases, but the problem is that the fix is in the env.py file that is in your migrations directory.

If you haven't made any changes to this file you can just take the updated version from [here](https://github.com/miguelgrinberg/Flask-Migrate/blob/main/src/flask_migrate/templates/flask/env.py) and drop it in place of the older one.

If you prefer a cleaner approach, then rename your migrations folder to some other name, then create a new migrations repository with `flask db init` after installing the latest Flask-Migrate, and then finally move all the Python scripts in the older migrations/versions directory to the one you just created.

**Comment by mokadevcloud** [comment]
Awesome, thanks for the quick response!
Couple of minor improvements possible in env.py:

1. add docstrings to get_engine() and get_engine_url()
2. get_engine() has the older version in the try block and the new version in the except block, while it is the opposite in get_engine_url(). I think we can consistently use the latter order.

Thanks again.

**Comment by miguelgrinberg** [comment]
I don't think it is worth making any changes at this point. The `get_engine()` will be removed in 3.1 and the deprecation warning will not appear anymore. If this warning bothers you, feel free to edit your env.py.

**Comment by mocaccano** [comment]
@emmanuelito As mentioned in my comment above, swapping the lines in the try and except blocks will remove the DeprecationWarning:

```
def get_engine():
    try:
        # this works with Flask-SQLAlchemy>=3
        return current_app.extensions["migrate"].db.engine
    except TypeError:
        # this works with Flask-SQLAlchemy<3 and Alchemical
        return current_app.extensions["migrate"].db.get_engine()
```
